curl/tests/client_test.cc

```diff
@@ -487,15 +491,30 @@ namespace leatherman { namespace curl {
              REQUIRE_THROWS_WITH(test_client.download_file(req, file_path), "easy perform failed"); 
 
              // Ensure that the temp file was removed
-             REQUIRE_THROWS_AS(find_matching_file(TEMP_FILE_REGEX), std::runtime_error);
+             REQUIRE(fs::is_empty(temp_dir_path));
          }
 
-         SECTION("when curl_easy_perform fails and removal of the temporary file fails, then both the errbuf and the failed temporary file removal message are contained in the thrown http_file_download_exception") {
+         SECTION("when curl_easy_perform fails and removal of the temporary file fails, then both the errbuf and the failed temporary file removal message are contained in the thrown http_file_operation_exception") {
              std::string file_path = (temp_dir_path / "file").string();
              curl::request req("");
              test_impl->test_failure_mode = curl_impl::error_mode::easy_perform_error; 
              test_impl->trigger_external_failure = setup_temp_removal_failure;
              REQUIRE_THROWS_WITH(test_client.download_file(req, file_path), Catch::StartsWith("easy perform failed and failed to remove temporary file file_util_fixture_")); 
          }
+
+         SECTION("when curl_easy_perform fails due to a CURLE_WRITE_ERROR, an http_file_operation_exception is thrown") {
```

Comment:
I'm wondering if for tests such as these, we should either:

(1) Write file_download in a try-catch block w/ an http_file_operation_exception in the catch, and have a statement in there that requires the message to match what we want or

(2) Create a custom matcher that ensures the thrown exception is the right type and it contains the message we want.

From what I've seen, Catch doesn't provide a matcher that checks for both, it's either you check the type of exception thrown OR the message but not both. Unless I'm mistaken.